.ci/test_r_package.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+
+# set up R environment
+R_LIB_PATH=~/Rlib
+mkdir -p $R_LIB_PATH
+echo "R_LIBS=$R_LIB_PATH" > ${HOME}/.Renviron
+echo 'options(repos = "https://cran.rstudio.com")' > ${HOME}/.Rprofile
+export PATH="$R_LIB_PATH/R/bin:$PATH"
+
+# installing precompiled R for Ubuntu
+# https://cran.r-project.org/bin/linux/ubuntu/#installation
+# adding steps from https://stackoverflow.com/a/56378217/3986677 to get latest version
+#
+# This only needs to get run on Travis because R environment for Linux
+# used by Azure pipelines is set up in https://github.com/guolinke/lightgbm-ci-docker
+if [[ $TRAVIS == "true" ]] &&  [[ $OS_NAME == "linux" ]]; then
+    sudo add-apt-repository \
+        "deb https://cloud.r-project.org/bin/linux/ubuntu bionic-cran35/"
+    sudo apt-key adv \
+        --keyserver keyserver.ubuntu.com \
+        --recv-keys E298A3A825C0D65DFD57CBB651716619E084DAB9
+    sudo apt-get update
+    sudo apt-get install \
+        --no-install-recommends \
+        -y \
+            r-base-dev=${R_TRAVIS_LINUX_VERSION} \
+            texlive-latex-recommended \
+            texlive-fonts-recommended \
+            texlive-fonts-extra \
+            qpdf \
+            || exit -1
+fi
+
+# Installing R precompiled for Mac OS 10.11 or higher
+if [[ $OS_NAME == "macos" ]]; then
+    wget -q https://cran.r-project.org/bin/macosx/R-${R_MAC_VERSION}.pkg -O R.pkg
+    sudo installer \
+        -pkg $(pwd)/R.pkg \
+        -target /
+
+    # Fix "duplicate libomp versions" issue on Mac
+    # by replacing the R libomp.dylib with a symlink to the one installed with brew
+    echo "fixing libomp stuff (compiler ${COMPILER})"
+    if [[ $AZURE == "true" ]]; then
+        sudo ln -sf \
+            "$(brew --cellar libomp)"/*/lib/libomp.dylib \
+            /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libomp.dylib
+    fi
```

Comment:
I think it's better to add `... || exit -1` than `echo`. Also, my assumption is that this workaround is needed only for `$COMPILER == "clang"`.

Reply:
ah I actually only had this `echo` in there for debugging! I actually meant to remove it.

I think you are right that it's only needing for `clang`, I can add that guard.